docs/10-lifecycle.md

```diff
@@ -0,0 +1,81 @@
+# Lifecycle
```

Comment:
I'm going to do any diagrams under a separate PR though. Not really 100% necessary for launch given that the existing docs don't have diagrams.